app/services/katello/pulp3/repository/apt.rb

```diff
@@ -0,0 +1,79 @@
+require 'pulp_deb_client'
+
+module Katello
+  module Pulp3
+    class Repository
+      class Apt < ::Katello::Pulp3::Repository
+        def remote_options
+          deb_remote_options = {
+            distributions: root.deb_releases
+          }
+          deb_remote_options[:components] = root.deb_components.present? ? root.deb_components : nil
+          deb_remote_options[:architectures] = root.deb_architectures.present? ? root.deb_architectures : nil
+
+          if root.url.blank?
+            deb_remote_options[:url] = nil
+          end
+
+          deb_remote_options[:gpgkey] = root.gpg_key.present? ? root.gpg_key.content : nil
+
+          common_remote_options.merge(deb_remote_options)
+        end
+
+        def publication_options(repository_version)
+          ss = api.signing_services_api.list(name: 'katello_deb_sign').results
+          popts = super(repository_version)
+          popts.merge!(
+            {
+              # structured is not necessary for subscription-manager
+              #structured: true, # publish real suites (e.g. 'stable')
+              simple: true # publish all into 'default'-suite
+            }
+          )
+          popts[:signing_service] = ss[0].pulp_href if ss && ss.length == 1
+          popts
+        end
+
+        def distribution_options(path)
+          {
+            base_path: path,
+            publication: repo.publication_href,
+            name: "#{generate_backend_object_name}"
+          }
+        end
+
+        def import_distribution_data
+          distribution = ::Katello::Pulp3::Distribution.fetch_content_list(repository_version: repo.version_href)
+          if distribution.results.present?
+            repo.update!(
+              :distribution_version => distribution.results.first.release_version,
+              :distribution_arch => distribution.results.first.arch,
+              :distribution_family => distribution.results.first.release_name,
+              :distribution_uuid => distribution.results.first.pulp_href,
+              :distribution_bootable => self.class.distribution_bootable?(distribution)
+            )
+            unless distribution.results.first.variants.empty?
+              unless distribution.results.first.variants.first.name.nil?
+                repo.update!(:distribution_variant => distribution.results.first.variants.first.name)
+              end
+            end
+          end
+        end
```

Comment:
i think this method can be deleted, this is related to rpm kickstart trees.

Reply:
Done, thanks